ext_mongodb.php

```diff
@@ -718,17 +775,82 @@ function __debugInfo() : array;
 
 final class Javascript implements Type, \Serializable
 {
-	use DenySerialization;
-
 	private $code;
 
-	public function __construct(string $code, private ?mixed $scope = NULL)
+	static private function checkArray(array $state)
+	{
+		if (
+			!array_key_exists( 'code', $state ) || !is_string( $state['code'] )
+		) {
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "MongoDB\BSON\Javascript initialization requires \"code\" string field" );
+		}
+
+		if ( array_key_exists( 'scope', $state ) && ( $state['scope'] !== NULL ) )
+		{
+			if ( !is_array( $state['scope'] ) && !is_object( $state['scope'] ) )
+			{
+				$valueType = gettype( $state['scope'] );
+				throw new \MongoDB\Driver\Exception\InvalidArgumentException( "Expected scope to be an array or object, {$valueType} given" );
+			}
+		}
+	}
+
+	public function serialize() : string
+	{
+		$s = [];
+		$s['code'] = $this->code;
+		if ( isset( $this->scope ) )
+		{
```

Comment:
Silly code formatting question: Do we need braces on newlines for `if`, `for`, etc.? I thought this was only for functions and classes. I realize you do that for the PHP.net code samples, but this is actual code.